### PR TITLE
Not ensuring you are on latest master leads to new submodules being e…

### PR DESCRIPTION
**Pull request by frankvanrietimqs**
…xcluded

### CODE REVIEW

File: create-new-rc.rb [hunk 1]
```diff
@@ -16,13 +16,13 @@
...
 # Create the RC branch
-discover_submodules(RootRepo).each { |sub|
+discover_submodules(RootRepo, "master").each { |sub|
 	print("Promoting #{sub} master to rc\n")
 	create_branch(RootRepo + "/" + sub, "master", "rc")
 }
 
 # Specify the release number and write it to "release number.txt"
-File.open(File.join(RootRepo,"release-number.txt", "w") do |f|
+File.open(File.join(RootRepo,"release-number.txt"), "w") do |f|
```
Comment (bmharper) [inline create-new-rc.rb]: I don't even....!

File: common.rb [hunk 2]
```diff
@@ -5,8 +5,11 @@
 ChannelsRoot = File.dirname(File.dirname(__FILE__)) + "/channels"
 VersionsRoot = File.dirname(File.dirname(__FILE__)) + "/versions"
 
-def discover_submodules(repo)
+def discover_submodules(repo, branch)
 	Dir.chdir(repo) {
+		exec_or_die("", "git checkout origin/#{branch}")
```
Comment (bmharper) [inline common.rb]: I guess this is OK for master, but I'm just afraid that somehow it's going to land us in trouble if we end up using this code other other branches which end up being rebased. For a rebased branch, the "git pull origin branch" is going to  try to do a merge, which will be a mess.
What would be more robust.. I think.. is this:

    git fetch origin #{branch}
    git checkout #{branch}
    git reset --hard origin/#{branch}

The proposed solution works even if `branch` is rebased.

File: create-new-rc.rb (hunk 1 above)
Comment (frankvanrietimqs) [inline create-new-rc.rb]: Yeah I had to fight the urge...

File: common.rb (hunk 2 above)
Comment (frankvanrietimqs) [inline common.rb]: Makes sense

**Pull request merged**